### ISSUE TITLE
K8s metadata -  deprecating a k8s version does not work

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#21695]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**

- run `Rancher-testing: v2.3.2`
- modify the metadata to set the highest minor version for v1.11 is `v1.11.8-rancher1-1`
- create a cluster with k8s `v1.11.8-rancher1-1`
- upgrade the metadata to add `v1.11.9-rancher1-2`, and deprecate `v1.11.8-rancher1-1`
- refresh the kontainerdrivers in Rancher
- see that the k8s `v1.11.8-rancher1-1` is gone from the list
- edit the existing cluster to add a control plane node 

**Result:**
- the node is added and the cluster is updated successfully

**Expected Results:**

- We should NOT be able to add new etcd/ control plane node, 
- we should be able to upgrade the cluster's k8s to a higher version.

**Comment by jiaqiluo** [comment]
The same bug for deprecating a major k8s version

**Steps:**

- run `Rancher-testing: v2.3.2`
- set the metadata that v2.3.2 supports k8s v1.11.x
- add a cluster with k8s v1.11.9
- modify the metadata to deprecate k8s v1.11

```
"v1.11": {
			MaxRancherVersion: "2.3.1",
			MaxRKEVersion:     "0.2.2",
		},
		"v1.11.9-rancher1-1": {
			MaxRancherVersion: "2.3.2",
			MaxRKEVersion:     "0.2.2",
		},
		"v1.11.9-rancher1-2": {
			MaxRancherVersion: "2.3.2",
			MaxRKEVersion:     "0.2.2",
		},
```

- refresh the kontainerdrivers in Rancher
- edit the existing cluster to add 2 etcd nodes

**Result:**
- the node is added and the cluster is updated successfully

**Expected Results:**

- We should NOT be able to add new etcd/ control plane node, 
- we should be able to upgrade the cluster's k8s to a higher version.

**Comment by jiaqiluo** [comment]
Reopen the bug, failed to validate in the setup upgraded from v2.2.8 to v2.3.0-rc4

Steps:
- run Rancher:v2.2.8 single install
- add a cluster with k8s version `v1.14.6-rancher1-1`
- upgrade to Rancher:v2.3.0-rc4
- edit the rke-metadata-config to point to my metadata git repo
- deprecate the k8s version in the metadata
```
"v1.14.6-rancher1-1": {
            DeprecateRKEVersion:     "0.2.2",
            DeprecateRancherVersion: "2.2.9",
        },
```
- refresh metadata in Rancher, then see the new data applied 
```
name": "k8s-versions-deprecated",
"source": "db",
"type": "setting",
"uuid": "74fe183c-d975-11e9-a499-0242ac110002",
"value": "{\"v1.14.6-rancher1-1\":true}"
```
- edit the existing cluster 

Results:
- I can still add etcd/ control plane / worker nodes
- I can change the `Recurring etcd Snapshot Interval`

Both of the above actions will trigger `rke up` 

More info:
```
k8s versions in v2.2.8
- v1.12.10-rancher1-2
- v1.13.10-rancher1-2
- v1.14.6-rancher1-1
- v1.15.3-rancher1-1

k8s versions in v2.3.0-rc4
- v1.13.10-rancher1-3
- v1.14.6-rancher3-1
- v1.15.3-rancher3-1
- v1.16.0-beta.1-rancher2-1
```

**Comment by jiaqiluo** [comment]
Failed to validate in the setup upgraded from v2.2.8 to v2.3.0-rc6

The steps are the same my previous comment: https://github.com/rancher/rancher/issues/21695#issuecomment-532462634

The good news is that the version `v1.14.6-rancher1-1` is deprecated,
in `v3/settings` we see
```
"name": "k8s-versions-deprecated",
"source": "db",
"type": "setting",
"uuid": "510e2b89-db28-11e9-bef5-0242ac110002",
"value": "{\"v1.14.6-rancher1-1\":true}"
}
```
and if I edit the clsuter `do` it will go to the upgrading state with the following message:
```
Failed to find system images for version v1.14.6-rancher1-1, it is deprecated
```
<img width="1497" alt="Screen Shot 2019-09-20 at 10 03 46 AM" src="https://user-images.githubusercontent.com/6218999/65346818-7a97fd00-db92-11e9-8c22-b761d65485bc.png">


But there are 2 issues showing up 
1. the cluster `do` is stuck in the upgrading state forever
2. editing the `do` cluster causes another cluster `ec2` in the upgrading state. `ec2` and `do` have the same k8s version. 

<img width="1511" alt="Screen Shot 2019-09-20 at 10 14 39 AM" src="https://user-images.githubusercontent.com/6218999/65346998-e4b0a200-db92-11e9-8936-a72e5c3d048b.png">


In Rancher logs we see
```
2019/09/20 17:16:27 [ERROR] ClusterController c-m78q9 [cluster-provisioner-controller] failed with : failed to find system images for version v1.14.6-rancher1-1, it is deprecated
2019/09/20 17:16:40 http: TLS handshake error from 127.0.0.1:47722: EOF
2019/09/20 17:16:50 [ERROR] ClusterController c-bm6f8 [cluster-provisioner-controller] failed with : failed to find system images for version v1.14.6-rancher1-1, it is deprecated
```

**Comment by jiaqiluo** [comment]
Another testing scenario will be 
- run v2.3.0-rc1
- use the custom metadata that sets `v1.14.6-rancher1-1` is available in rc1 but deprecated in rc6  
- add 2 clusters with k8s `v1.14.6-rancher1-1`
- upgrade Rancher to rc6
- edit the cluster 

Expected:
- we cannot edit the cluster
- editing cluster1 should not affect cluster2, even they have the same k8s version.

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher:v2.3.0-rc9 upgraded from v2.2.8

Steps:
- run `Rancher:v2.2.8` single install
- add a cluster with k8s version `v1.14.6-rancher1-1`
- upgrade to `Rancher:v2.3.0-rc9`
- edit the rke-metadata-config to point to my metadata git repo
- deprecate the k8s version in the metadata
```
"v1.14.6-rancher1-1": {
            DeprecateRKEVersion:     "0.2.2",
            DeprecateRancherVersion: "2.2.9",
        },
```
- refresh metadata in Rancher, then see the new data applied 
```
name": "k8s-versions-deprecated",
"source": "db",
"type": "setting",
"uuid": "74fe183c-d975-11e9-a499-0242ac110002",
"value": "{\"v1.14.6-rancher1-1\":true}"
```
- edit the existing cluster 

Results:

the UI prevents me from saving any changes to the cluster 
<img width="1505" alt="Screen Shot 2019-09-25 at 10 51 10 AM" src="https://user-images.githubusercontent.com/6218999/65626997-4736e300-df83-11e9-8b4a-207adb7ce928.png">


Note:

Now the behavior is that user cannot edit the cluster but can scale up/down nodes from the cluster’s node page
